### PR TITLE
[BugFix] Forbid creating materialized view using week/quarter/year to refresh

### PR DESCRIPTION
**Pull request by zddr**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes # https://github.com/StarRocks/starrocks/issues/9446

### CODE REVIEW

**Comment by zddr** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 14 / 16 (87.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|14|16|87.50%|[1035, 1117]|

**Review by Astralidea — DISMISSED** [review]

**Comment by Astralidea** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 1]
```diff
@@ -1025,6 +1025,19 @@ public ParseNode visitCreateMaterializedViewStatement(
             }
             return new CreateMaterializedViewStmt(tableName.getTbl(), queryStatement, properties);
         }
+        if (refreshSchemeDesc instanceof AsyncRefreshSchemeDesc) {
+            AsyncRefreshSchemeDesc asyncRefreshSchemeDesc = (AsyncRefreshSchemeDesc) refreshSchemeDesc;
+            if (asyncRefreshSchemeDesc.getIntervalLiteral() != null &&
+                    asyncRefreshSchemeDesc.getIntervalLiteral().getUnitIdentifier() != null) {
+                String description = asyncRefreshSchemeDesc.getIntervalLiteral().getUnitIdentifier().getDescription();
+                if (!("SECOND".equalsIgnoreCase(description) || "MINUTE".equalsIgnoreCase(description) ||
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1033]: extract this logic into a function and call it in both create and alter.
And more, push supported units into a unmodifiable set and just use find to judge whether the unit is supported.
And more, add an UT for this change.

**Review by ABingHuang — CHANGES_REQUESTED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 16 / 18 (88.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|16|18|88.89%|[1008, 1128]|

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 2]
```diff
@@ -1094,6 +1122,13 @@ public ParseNode visitAlterMaterializedViewStatement(
         if (context.refreshSchemeDesc() != null) {
             refreshSchemeDesc = ((RefreshSchemeDesc) visit(context.refreshSchemeDesc()));
         }
+        if (refreshSchemeDesc instanceof AsyncRefreshSchemeDesc) {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1125]: add an UT for alter materialized view

**Review by ABingHuang — CHANGES_REQUESTED** [review]
add an UT for alter materialized view

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 17 / 18 (94.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|17|18|94.44%|[1008]|

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]
LGTM

**Review by ABingHuang — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Pull request merged**